### ISSUE TITLE
'`GLIBC_2.34' not found' error after upgrade 12.0.0 on Linux OS

### ISSUE THREAD

**Issue by jooy2** [realm/realm-js#6082]
### How frequently does the bug occur?

Always

### Description

Hello,
Thank you for developing realm. After upgrading to the recently released version 12.0.0, I started receiving the following error message.

(see below section)

The version immediately before, 11.10.2, had no issues.
I'm using realm in NodeJS 16.17 + Electron 25, and I'm using Linux Ubuntu 18.04 LTS.

However, this issue also occurred with the Ubuntu 20.04 LTS + NodeJS 18.17.1 combination. Also the problem occurs both with and without database encryption.

Is there a solution to this?

Thanks.

### Stacktrace & log output

```shell
Uncaught Error: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /home/user/Documents/source/myProject/node_modules/realm/generated/ts/realm.node)
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:2:1822)
    at Module._extensions..node (node:internal/modules/cjs/loader:1354:18)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:2:1822)
    at Module.load (node:internal/modules/cjs/loader:1124:32)
    at Module._load (node:internal/modules/cjs/loader:965:12)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
    at o._load (node:electron/js2c/renderer_init:2:3109)
    at Module.require (node:internal/modules/cjs/loader:1148:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/home/user/Documents…ndle.node.js:110:22)
```

### Can you reproduce the bug?

Always

### Reproduction Steps

This is an internal project, so I can't post the code I went through, but I'm attaching the project repository that I reproduced with the super simple electron template below:

https://github.com/jooy2/issue-realm-js-12-glibc

```shell
$ git clone https://github.com/jooy2/issue-realm-js-12-glibc.git
$ npm i
$ npm run start
```

See the console error in chrome devtools:

![image](https://github.com/realm/realm-js/assets/48266008/fa0adf86-7546-428d-b8e1-b6b8e2ffe4b7)

### Version

12.0.0

### What services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

Linux Ubuntu 18.04 / 20.04 LTS

### Build environment

Electron: 25 or 26
NodeJS: 18.17.1 or 16.16.0

**Comment by kneth** [comment]
@jooy2 Thank you for reporting. While we investigate, you can try some of the suggestions in https://stackoverflow.com/questions/71940179/error-lib-x86-64-linux-gnu-libc-so-6-version-glibc-2-34-not-found

**Comment by jooy2** [comment]
Thank you for your quick response.

Upgrading glibc manually from previous Ubuntu releases was beyond my capabilities, as I've always had problems upgrading glibc manually.
This is the currently installed version of glibc on Linux (Ubuntu 20.04), which does not work.

```shell
$ ldd --version | head -n1
ldd (Ubuntu GLIBC 2.31-0ubuntu9.9) 2.31
```

Here's one thing I've tried, and I've found that it works fine on the Ubuntu 22.04 LTS + NodeJS 18.17.1 combination.

![image](https://github.com/realm/realm-js/assets/48266008/81c62194-f7bb-4600-8027-33bd6df5d2e3)

It was showing version 2.35 of glibc.

```shell
test@test-virtual-machine:~/Documents/source/issue-realm-js-12-glibc$ ldd --version | head -n1
ldd (Ubuntu GLIBC 2.35-0ubuntu3.1) 2.35
```

This issue has a significant impact on my production. I will be using realm v11 until then, but I hope this issue is resolved quickly.

Regards,
